frontend/packages/dev-console/src/components/topology/filters/TopologyFilterBar.tsx

```diff
@@ -0,0 +1,61 @@
+import * as React from 'react';
+import { connect, Dispatch } from 'react-redux';
+import { Toolbar, ToolbarGroup, ToolbarItem } from '@patternfly/react-core';
+import { RootState } from '@console/internal/redux';
+import { getTopologyFilters } from '@console/internal/reducers/ui';
+import { setTopologyFilters } from '@console/internal/actions/ui';
+import { TextFilter } from '@console/internal/components/factory';
+import FilterDropdown from './FilterDropdown';
+import { TopologyFilters, DisplayFilters } from './filter-utils';
+import './TopologyFilterBar.scss';
+
+type StateProps = {
+  filters: TopologyFilters;
+};
+
+type DispatchProps = {
+  onFilterChange: (filter: TopologyFilters) => void;
+};
+
+type TopologyFilterBarProps = StateProps & DispatchProps;
+
+const TopologyFilterBar: React.FC<TopologyFilterBarProps> = ({ filters, onFilterChange }) => {
+  const { display, searchQuery } = filters;
+  const onDisplayFilterChange = (displayFilters: DisplayFilters) =>
+    onFilterChange({ display: displayFilters, searchQuery });
+
+  const onSearchQueryChange = (e: React.KeyboardEvent) =>
+    onFilterChange({
+      display,
+      searchQuery: (e.target as HTMLInputElement).value,
+    });
```

Comment:
At first, I wondered why wouldn't we allow updating of the redux state filters individually and let the reducer merge the values instead of requiring us to send the full filters payload each time.
Right now there isn't much benefit in doing it that way so I suppose this is ok for now.
Thoughts @andrewballantyne